Code/GraphMol/MarvinParse/MarvinDefs.cpp

```diff
@@ -0,0 +1,2978 @@
+          centerOfGroup.x / marvinSuperatomSgroup->atoms.size();
+      dummyParentAtom->y2 =
+          centerOfGroup.y / marvinSuperatomSgroup->atoms.size();
+    }
+  }
+
+  // move any children from the old expanded group to the new superatomSgroup
+
+  for (auto sgroupPtr : this->sgroups) {
+    moveSgroup(sgroupPtr, marvinSuperatomSgroup, false);
+  }
+  this->sgroups.clear();
+
+  // remove the old expanded superatom sgroup
+
+  parent->sgroups.erase(
+      std::find(parent->sgroups.begin(), parent->sgroups.end(), this));
+  delete this;
+
+  // fix up any siblings that contain all the atoms of this group.
+
+  for (auto sibling : marvinSuperatomSgroup->parent->sgroups) {
+    if (sibling == marvinSuperatomSgroup) {
+      continue;  // not a sibling
+    }
+
+    // see if the new superatom group has all of its atoms in the sibling.  If
+    // to add the dummy atom to the sibling and remove the atoms (there must
+    // be at least one atom in the sibling that is not in the new superatom,
+    // or it would be a child of the superatom already
+
+    auto isSgroupInSetOfAtomsResult =
+        marvinSuperatomSgroup->isSgroupInSetOfAtoms(sibling->atoms);
+
+    switch (isSgroupInSetOfAtomsResult) {
+      case SgroupInAtomSet:
+        // remove the group atoms from the sibling and add the dumm atom
+
+        sibling->atoms.push_back(dummyParentAtom);
+        for (auto atomToRemove : marvinSuperatomSgroup->atoms) {
+          int index = sibling->getAtomIndex(atomToRemove->id);
+          sibling->atoms.erase(sibling->atoms.begin() + index);
+        }
+
+        break;
+
+      case SgroupNotInAtomSet:
+        // do nothing
+        break;
+
+      case SgroupBothInAndNotInAtomSet:
+        throw FileParseException(
+            "a subgroup contains some of the atoms of a superatomSgroup - this should not happen");
+    }
+  }
+
+  return marvinSuperatomSgroup;
+}
+
+int MarvinMultipleSgroup::getMatchedOrphanBondIndex(
+    std::string atomIdToCheck, std::vector<MarvinBond *> &bondsToTry,
+    std::vector<MarvinBond *> &orphanedBonds) const {
+  for (auto testBond = bondsToTry.begin(); testBond != bondsToTry.end();
+       ++testBond) {
+    if (*testBond == NULL) {
+      continue;
+    }
+    std::string otherAtomId;
+    if ((*testBond)->atomRefs2[0] == atomIdToCheck) {
+      otherAtomId = (*testBond)->atomRefs2[1];
+    } else if ((*testBond)->atomRefs2[1] == atomIdToCheck) {
+      otherAtomId = (*testBond)->atomRefs2[0];
+    } else {
+      continue;  // bond does not have the atomId
+    }
+
+    auto testBondIter =
+        find(orphanedBonds.begin(), orphanedBonds.end(), *testBond);
+    if (testBondIter != orphanedBonds.end()) {
+      return testBondIter - orphanedBonds.begin();
+    }
+
+    *testBond = NULL;
+
+    // try the children
+
+    int childId =
+        getMatchedOrphanBondIndex(otherAtomId, bondsToTry, orphanedBonds);
+    if (childId >= 0) {
+      return childId;
+    }
+  }
+
+  return (-1);
```

Comment:
why the parens?

Reply:
this is an old habit.  Decades ago I chased a bug for days that was written in C as:
val =-1;

Kernigan and Ritchie made the operator "=-" the same as "-=", so val=-1 is different than val = -1.  Since then I have always put -1 in parentheses.   It does nothing here, so I removed it.   As far as I can tell, c++ does not have a -= operator.  Most C compilers flag it as an error now.  I also removed it in three other places.